File: source/Emphasize/Format Definitions/Code.vb
Imports Microsoft.VisualStudio.Text.Classification
Imports Microsoft.VisualStudio.Utilities
Imports System.ComponentModel.Composition
Imports System.Windows.Media


Namespace FormatDefinitions

    <ClassificationType(ClassificationTypeNames:=Code.Name)>
    <Export(GetType(EditorFormatDefinition))>
    <Name(Code.Name)>
    <Order(After:=Priority.High)>
    <UserVisible(True)>
    Public Class Code
        Inherits ClassificationFormatDefinition


        Public Const Name As String = "Comment - Code Span"


        <Export(GetType(ClassificationTypeDefinition))>
        <BaseDefinition("Comment")>
        <Name(Name)>
        Public Shared ReadOnly ClassificationType As ClassificationTypeDefinition


        Public Sub New()
            DisplayName = Name
            BackgroundBrush = CreateBackgroundBrush()
            BackgroundOpacity = GetBackgroundOpacity()
        End Sub


        Friend Shared Function CreateBackgroundBrush() As Brush
            Return New SolidColorBrush(Color.FromRgb(0, 128, 0))
        End Function


        Friend Shared Function GetBackgroundOpacity() As Double
            Return 0.1
        End Function

    End Class

End Namespace

File: source/Emphasize/FormatDefinitions/Code.cs
#nullable enable

using Microsoft.VisualStudio.Text.Classification;
using Microsoft.VisualStudio.Utilities;
using System.ComponentModel.Composition;
using System.Windows.Media;


namespace Emphasize.FormatDefinitions {

    [ClassificationType(ClassificationTypeNames = Name)]
    [Export(typeof(EditorFormatDefinition))]
    [Name(Name)]
    [Order(After = Priority.High)]
    [UserVisible(true)]
    public class Code : ClassificationFormatDefinition {

        public const string Name = "Emphasize - Code";


        [Export(typeof(ClassificationTypeDefinition))]
        [Name(Name)]
        public static readonly ClassificationTypeDefinition? ClassificationType;


        public Code() {
            DisplayName = Name;
            BackgroundBrush = CreateBackgroundBrush();
            BackgroundOpacity = GetBackgroundOpacity();
        }


        internal static Brush CreateBackgroundBrush() {
            return new SolidColorBrush(Color.FromRgb(128, 128, 128));
        }


        internal static double GetBackgroundOpacity() {
            return 0.15;
        }

    }

}
